components/clarinet-deployments/src/onchain/mod.rs

```diff
@@ -687,7 +690,31 @@ pub fn apply_on_chain_deployment(
     // Phase 2: we submit all the transactions previously encoded,
     // and wait for their inclusion in a block before moving to the next batch.
     let mut current_block_height = 0;
-    for batch in batches.into_iter() {
+    for (epoch, batch) in batches.into_iter() {
+        // Ensure we've reached the appropriate epoch for this batch
+        let after_block = match epoch {
+            EpochSpec::Epoch2_05 => network_manifest.devnet.as_ref().unwrap().epoch_2_05,
+            EpochSpec::Epoch2_1 => network_manifest.devnet.as_ref().unwrap().epoch_2_1,
+        };
```

Comment:
the `apply_onchain_deployment` is also called by the command 
```bash
clarinet deployments apply --mainnet
```
and 
```bash
clarinet deployments apply --testnet
```

which would be an issue with this approach.

Reply:
Would it make sense to developers if clarinet just ignores the specified epoch when applying a deployment plan? I think just reporting a warning if epoch is specified would be reasonable. Something like:

```
Warning: `epoch` specified for a deployment batch is ignored when applying a deployment plan. This field should only be specified for deployments plans used to launch a devnet with `clarinet integrate`.
```

Reply:
works for me!